opm/parser/eclipse/EclipseState/Schedule/MSW/Valve.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+  Copyright 2019 Equinor.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef VALVE_HPP_HEADER_INCLUDED
+#define VALVE_HPP_HEADER_INCLUDED
+
+#include <map>
+#include <utility>
+#include <vector>
+
+
+namespace Opm {
+
+    class DeckRecord;
+    class DeckKeyword;
+    class Segment;
+
+    class Valve {
+    public:
+        enum class Status {
+            OPEN,
+            SHUT
+        };
+
+        explicit Valve(const DeckRecord& record);
+
+        // the function will return a map
+        // [
+        //     "WELL1" : [<seg1, valv1>, <seg2, valv2> ...]
+        //     ....
+        static std::map<std::string, std::vector<std::pair<int, Valve> > >
+        fromWSEGVAVLV(const DeckKeyword& keyword);
+
+        Status status() const;
+
+        // parameters for constriction pressure loss
+        double conFlowCoefficient() const;
+        double conCrossArea() const;
+        double conMaxCrossArea() const;
+
+        // parameters for pressure loss along the pipe
+        double pipeAdditionalLength() const;
+        double pipeDiameter() const;
+        double pipeRoughness() const;
+        double pipeCrossArea() const;
+
+        void setPipeDiameter(const double dia);
```

Comment:
I know `const` here does not do anything in the declaration. Any opinion whether we should bother to have it or not?

Reply:
Unlike return values, const function arguments prevent the implementation from changing the input argument, which is why I like it in general (preventing some nasty surprises).